src/guide/javascript.md

```diff
@@ -191,34 +207,135 @@ const kp = generateKeyPair({
-described above, to use transactions or queries you'll need to have a
-`keyPair` parameter as a part of the `Client` instance definition:
+When you have a key pair, you might create a `Signer` using the key pair :
 
 ```ts
-import { Client } from '@iroha2/client'
+import { KeyPair } from '@iroha2/crypto-core'
+import { Signer } from '@iroha2/client'
+import { AccountId, DomainId } from '@iroha2/data-model'
 
-const client = new Client({
-  torii: {
-    // Both URLs are optional in case you only need one of them,
-    // e.g. only the telemetry endpoints
-    apiURL: 'http://127.0.0.1:8080',
-    telemetryURL: 'http://127.0.0.1:8081',
-  },
-  accountId: AccountId({
-    // Account name
-    name: 'alice',
-    // The domain where this account is registered
-    domain_id: DomainId({
-      name: 'wonderland',
-    }),
+// Key pair from previous step
+declare const keyPair: KeyPair
+
+const accountId = AccountId({
+  // Account name
+  name: 'alice',
+  // The domain where this account is registered
+  domain_id: DomainId({
+    name: 'wonderland',
   }),
-  // A key pair, needed for transactions and queries
-  keyPair: kp,
 })
+
+const signer = new Signer(accountId, keyPair)
+```
+
+Well, now we are able to make signatures with `signer.sign(binary)`! But
+usually we need to interact with Iroha and send something to it, like
+transactions or queries. `Torii` will help us.
+
+`Torii` class represents HTTP / WebSocket communications with Iroha. It
+will help us to communicate with each Iroha Endpoint, i.e.:
+
+- Submit transactions
+- Make queries
+- Listen for events
+- Listen for blocks stream
+- and so on
+
+To initialize `Torii`, first we need to know Iroha Torii URLs. Our Iroha
+Peer is configured to listen for API endpoints at `http://127.0.0.1:8080`
+and for telemetry endpoints at `http://127.0.0.1:8081`. Then, we need to
+provide appropriate HTTP / WebSocket adapters which `Torii` will use[^1].
+These adapters depends on the environment you are going to use
+`@iroha2/client` in.
+
+[^1]:
+    We have to pass environment-specific `ws` and `fetch`, because there is
+    no way for Iroha Client to communicate with a peer in an
+    environment-agnostic way.
+
+In Node.js `Torii` initialization will look like this:
+
+```ts
+import { Torii } from '@iroha2/client'
+import { adapter as WS } from '@iroha2/client/web-socket/node'
+
+import nodeFetch from 'node-fetch'
+
+// another alternative
+import { fetch as undiciFetch } from 'undici'
+
+const torii = new Torii({
+  apiURL: 'http://127.0.0.1:8080',
+  telemetryURL: 'http://127.0.0.1:8081',
+  ws: WS,
+  fetch: nodeFetch as typeof fetch,
+})
+```
+
+::: tip
+
+In the example above, we use
+[`node-fetch`](https://www.npmjs.com/package/node-fetch) package which
+implements [Fetch API](https://fetch.spec.whatwg.org/#fetch-method) in
+Node.js. However, you can use
+[`undici`](https://undici.nodejs.org/#/?id=undicifetchinput-init-promise)
+as well.
+
+:::
```

Comment:
I think we should say about two alternatives before the code snippet, not after. 

1) One possibility: split the snippet in two parts, imports and torii initialization. This can be done with an ordered list. And when you introduce imports (item 1 on the list), you say: hey, there are two ways to go about it, and provide 2 separate snippets. Then item 2 on the list is torii init.

2) Or you can provide two complet4e snippets. Maybe one of them can be hidden via `::: details`

Reply:
I think it's not really important to explicitly show how to use `undici` and just mentioning it is enough.